ddtrace/debugging/_probe/model.py

```diff
@@ -79,68 +79,77 @@ def update(self, other: "Probe") -> None:
         if self.version == other.version:
             return
 
-        for attrib in (_.name for _ in self.__attrs_attrs__ if _.eq):
+        for attrib in (f.name for f in self.__dataclass_fields__.values() if f.compare):
```

Comment:
I think `dataclasses.fields(self)` would be equivalent to `self.__dataclass_fields__.values()`